computation_graph/graph.py

```diff
@@ -123,3 +124,46 @@ def unbound_signature(
         kwargs=tuple(keep_not_in_bound_kwargs(node.signature.kwargs)),
         optional_kwargs=tuple(keep_not_in_bound_kwargs(node.signature.optional_kwargs)),
     )
+
+
+def replace_source(x, y):
+    return _transform_edges(_edge_source_equals(x), _replace_edge_source(y))
+
+
+def _transform_edges(query, edge_mapper):
+    return _operate_on_subgraph(
+        _split_by_condition(query), gamla.compose(tuple, gamla.map(edge_mapper))
+    )
+
+
+def _edge_source_equals(x):
+    if not isinstance(x, base_types.ComputationNode):
+        x = make_computation_node(x)
```

Comment:
This if statement is redundant, it's already done at the beginning of `make_computation_node`.
```
def _edge_source_equals(x):
    x = make_computation_node(x)
    return lambda edge: edge.source == x
```

Reply:
I think we use it for both types, but maybe we can change it.